File: 0041-first-missing-positive/0041-first-missing-positive.java
class Solution {
    public int firstMissingPositive(int[] nums) {
        
        int[] result = new int[nums.length+1];
        
        int i = 0;
        while (i < nums.length) {
            if (nums[i] > nums.length || nums[i] < 1) {
                i++;
                continue;
            }
            result[nums[i]] += 1;
            i++;
        }
        
        i = 1;
        while (i < result.length && result[i] != 0) {
            i++;
        }
        return i;
        
    }
}

File: 0041-first-missing-positive/0041-first-missing-positive.py
class Solution:
    def firstMissingPositive(self, nums: List[int]) -> int:
        
        numsSet = set(nums)
        
        for i in range(1, len(nums)+1):
            if i not in numsSet:
                return i
        
        return len(nums)+1
        